src/arrayMethodJoin.js

```diff
@@ -4,8 +4,41 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+    let addToString = '';
+    let sep = separator;
+
+    if (this.length === 0) {
+      return result;
+    }
+
+    if (separator === null) {
+      sep = 'null';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      if (i === this.length - 1) {
```

Comment:
write the if statement in the opposite way:
add a separator if it's not the last element

In such a case nested if will be unnecessary